manubot/cite/curie/__init__.py

```diff
@@ -0,0 +1,106 @@
+"""
+Compact Uniform Resource Identifiers
+
+References:
+
+- https://en.wikipedia.org/wiki/CURIE
+- https://identifiers.org/
+- https://github.com/manubot/manubot/issues/218
+- https://docs.identifiers.org/articles/api.html
+- https://n2t.net/e/compact_ids.html
+- https://n2t.net/e/cdl_ebi_prefixes.yaml
+- https://en.wikipedia.org/wiki/MIRIAM_Registry
+- [Identifiers.org and MIRIAM Registry: community resources to provide persistent identification](https://doi.org/10.1093/nar/gkr1097)
+- [On the road to robust data citation](https://doi.org/10.1038/sdata.2018.95)
+
+"""
+import functools
+import json
+import logging
+import pathlib
+import re
+from typing import List, Set
+
+import requests
+
+_keep_namespace_fields = {
+    "prefix",
+    "mirId",  # MIRIAM Registry identifier
+    "name",
+    "pattern",  # regex pattern
+    "description",
+    "sampleId",  # example identifier
```

Comment:
Another key you should keep is `namespaceEmbeddedInLui`. This signifies namespaces whose pattern starts with the prefix. For example, the GO pattern is `^GO:\d{7}$`, but to get a proper resolution you should only write it once like https://identifiers.org/GO:0006915 and *not like* https://identifiers.org/GO:GO:0006915

Reply:
Thanks. Excluding `namespaceEmbeddedInLui` was a crucial oversight. Unfortunately, even with knowledge of `namespaceEmbeddedInLui`, I'm worried it's insufficient to resolve all CURIEs with case-insensitive prefixes as per my comments in:

- https://github.com/identifiers-org/identifiers-org.github.io/issues/100#issuecomment-614679142
- https://github.com/identifiers-org/identifiers-org.github.io/issues/99#issuecomment-614690283